python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -380,10 +423,31 @@ async def take_images(self, image_type):
 
         exposures = await self.take_image_type(image_type, exp_times)
 
-        await asyncio.wait_for(
-            self.image_in_oods_received_all_expected.wait(),
-            timeout=self.config.oods_timeout,
-        )
+        # Wait for image in OODS. Try several times before
+        # giving up, if there's a timeout.
+        do_wait_for_image_in_oods = True
+        retry = 0
+        retry_rate = self.config.oods_timeout_retry_rate
+        max_retry = self.config.oods_timeout_max_retry
+        while do_wait_for_image_in_oods:
```

Comment:
Uhm.. so you are doing a "timeout loop" to "timeout" waiting for the events to arrive. I don't think this is a good idea. 

Instead, consider improving the error message such that it tells us what we are missing. 

For example, see what I did in this ticket branch that helped me figure out the problem:

https://github.com/lsst-ts/ts_externalscripts/blob/aabbb62e04a254643e527ca0c34425372a3cadca/python/lsst/ts/externalscripts/base_make_calibrations.py#L404-L423